### PR TITLE
Auth url in query string bug fix

### PR DESCRIPTION
**Pull request by ikbalkaya**
This PR attempts to fix a bug that is caused by ably-java mis-reading `authUrl` when is provided through query string.
Originally spotted https://github.com/ably/ably-flutter/pull/466

Fixes https://github.com/ably/ably-java/issues/935

### CODE REVIEW

File: lib/src/main/java/io/ably/lib/rest/Auth.java [hunk 1]
```diff
@@ -805,6 +805,7 @@ public Object handleResponse(HttpCore.Response response, ErrorInfo error) throws
                 /* append all relevant params to token params */
                 Map<String, Param> urlParams = null;
                 URL authUrl = HttpUtils.parseUrl(authOptions.authUrl);
+                final String querylessUrl = HttpUtils.removeQueryFromURL(authOptions.authUrl);
```
Comment (sacOO7) [inline lib/src/main/java/io/ably/lib/rest/Auth.java:808]: can be renamed to `final String cleanUrlWithoutQueryParams`

File: lib/src/main/java/io/ably/lib/http/HttpUtils.java [hunk 2]
```diff
@@ -78,6 +80,28 @@ public static URL parseUrl(String url) throws AblyException {
...
+    /**
+     * Remvoes queery string from a url string and return the url string with the new url string
+     * @param url Url string that needs query string part removed
+     *
+     * @return  Url string with query string part removed, if existed in the first place
+     *
+     * @throws AblyException  built from URISyntaxException if java.net.URI fails to build
+     * the URI given url
+     * */
+    public static String removeQueryFromURL(String url) throws AblyException {
```
Comment (sacOO7) [inline lib/src/main/java/io/ably/lib/http/HttpUtils.java:92]: `removeQueryParamsFromURL`

File: lib/src/main/java/io/ably/lib/rest/Auth.java (hunk 1 above)
Comment (ikbalkaya) [reply lib/src/main/java/io/ably/lib/rest/Auth.java]: Thanks. I made names a bit more clear : dd09c50d5e7ecbb1e8a44aeea03947df84623763

File: lib/src/main/java/io/ably/lib/http/HttpUtils.java (hunk 2 above)
Comment (ikbalkaya) [reply lib/src/main/java/io/ably/lib/http/HttpUtils.java]: dd09c50d5e7ecbb1e8a44aeea03947df84623763

Comment (ikbalkaya) [reply lib/src/main/java/io/ably/lib/http/HttpUtils.java]: Resolving on behalf

**Review by owenpearson — APPROVED** [review]
LGTM, thanks

**Pull request merged**